openedx/openedx-platform
# YONK-776: added thread_followed and thread_unfollowed signals

This change includes newly added signals thread_followed and thread_unfollowed and their unit tests. Currently, this change can't be testing as we have no use case of this here. We are using these signals in our Solutions Fork as we have to track these two. Adding this here so that anyone from the community can use it if want to perform any custom task on thread follow or unfollow feature.  

**UPDATE**: Added thread_followed and thread_unfollowed receivers in Team Discussions [(models.py#L49).](https://github.com/edx/edx-platform/blob/7c3dab08a7fe2e26f21d91aa6475d2bd574ae0e8/lms/djangoapps/teams/models.py#L49) These receivers log the activity of the user, whenever these actions are performed. 

**Business Purpose**
Last activity should be updated in case of un-voting, following or unfollowing any post as these actions should also reflect that some activity is performed in any specific thread. Currently, it is not updating on the mentioned actions.

**How to test?**
I have created a Sandbox [aamishbaloch.sandbox.edx.org](https://aamishbaloch.sandbox.edx.org). User a get in with the newly created account and enroll himself in the DemoX edX Demonstration Course. The next step is to join the team, say My Team X (already created in that course) and then perform any activity of follow or unfollow any post. You will see that the Last activity time will also be updated on the actions mentioned above.
Before this Update, Last activity time was not updating on follow or unfollow activities.
